crates/fuel-streams-core/src/stream/stream_impl.rs

```diff
@@ -199,28 +224,101 @@ impl<S: Streamable> Stream<S> {
         self.store.stream_name.as_str()
     }
 
-    // TODO: This should probably be `subscribe_raw` since it returns pure bytes
+    // Less performant due to our hybrid use of NATS and S3
+    pub async fn subscribe_raw(
+        &self,
+        subscription_config: Option<SubscriptionConfig>,
+    ) -> Result<BoxStream<'_, Vec<u8>>, StreamError> {
+        let mut config = PullConsumerConfig {
+            filter_subjects: vec![S::WILDCARD_LIST[0].to_string()],
+            deliver_policy: DeliverPolicy::All,
```

Comment:
```suggestion
            filter_subjects: subscription_config.unwrap_or(vec![S::WILDCARD_LIST[0].to_string()]),
            deliver_policy: subscription_config.unwrap_or(DeliverPolicy::All),
```

Reply:
Here we `unwrap` twice, and as the config grows we will keep unwrapping. Mutating after the default would be more maintainable here. 

Also, most of these user-exposed functions here need some re-visiting. we shouldn't have these many `subscribe` functions but since that's the current state, I'd suggest we tackle those in a well-refined task